### ISSUE TITLE
New WinDBG is no longer on the Store.

### ISSUE THREAD

**Issue by sylveon** [microsoft/WinDbg-Feedback#91]
[This page here](https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/) mentions the following:
> To get and stay on the latest release, install WinDbg as described on this page. WinDbg Preview will not receive further updates in the Microsoft Store.

This is very disappointing to me as I use the Store as an easy way to acquire WinDBG and keep it up to date. Why this decision? Couldn't the final non-preview also be on there? It's an .msix file too so it should be trivial to keep a copy on the Store as well.

**Comment by sylveon** [comment]
> No one who installed WinDbg via the Store will know that though, because weirdly enough, people don't go to the store listings for apps they already have installed

Yes, that's my biggest problem. One of the initial reason for the new WinDBG was automatically updating people on old versions, as mentioned [here](https://github.com/microsoftfeedback/WinDbg-Feedback/issues/19#issue-457758964). Everybody which installed the Store version now is back to square one regarding this, with no indication that they're on an unsupported version now. Hell, I only found out because somebody linked a tweet to me saying so. The up-to-date store version of WinDBG Preview and its listing is still doing business as usual, not mentioning how it's deprecated now.

Worse, unless the listing is made private, people might believe that WinDBG Preview on the store to be a beta version for the stable release, when this isn't true.

Currently, to install the new version I have to dig out a documentation page that is at the *very bottom* of the Google results for "install WinDBG", download a file, open it, hit install, and then it's installed.

Compare that to the store being open store > search "windbg" > hit install. It's a clear downgrade. I'm not against options as I recognize that corporate environments block the store, but I feel like the current solution (appinstaller/msix) will not please anyone. It actively makes the experience worse for people who do/can use the store, and in the cases where the store is blocked/missing, there's a *high* chance that IT also blocked App Installer...

Hell, I have such an app with store and appinstaller versions, and when I point people for which the store doesn't work to the appinstaller version, it 90% of the time also does not work.

In my experience, the background auto update system isn't as seamless on appinstaller than it is on the store either.

**Comment by roxk** [comment]
Second windbg should be on the store. Just do it like WSA: put it on the store, but also provide msix for ppl to install manually if they want. Easy Win-Win.

**Comment by dongle-the-gadget** [comment]
Actually I don't think WSA provides MSIX outside the Store. Windows Terminal would be a better example (they recommending people use Store, but they also provide MSIX in GitHub Releases).

**Comment by jcotton42** [comment]
Having the debugger available in the Store is incredibly convenient. I've walked people through installing and running it for deeper troubleshooting, and being able to install it from the Store makes the process that much easier.

**Comment by lzybkr** [comment]
We do appreciate the feedback and will try to be responsive.

If appinstaller isn't working for enough folks, we'll explore alternatives.

One reason we planned to drop store releases is to save on engineering effort. That is currently a manual process and doesn't happen as often as we release internally. We can always revisit this decision with appropriate feedback.

Note that as of Friday last week, you can install with the command line `winget install Microsoft.WinDbg` - that seems even more friendly than searching the store.

Also note - we have one more store release planned which matches the current release other than pointing folks to https://aka.ms/windbg/download for future updates. We couldn't publish that version before having the download version available.

**Comment by dongle-the-gadget** [comment]
WinGet doesn't support automatic updates, unfortunately.

**Comment by MV10** [comment]
Since #19 was closed (!) ... Neither the Store nor a stand-alone installer (particularly one needing admin rights and Internet access!) is appropriate for locked-down corporate environments. These changes make it nearly impossible for us to obtain a copy without elaborate work-arounds.

**Comment by sylveon** [comment]
This topic is purely about a store version of windbg. The ask for a portable zip version can go in its own issues.

**Comment by agowa338** [comment]
It was really easy to find windbg in the Windows store for clients. The only issue that needed to be solved was getting it on a Windows server, as the Windows store did not exist there. That was it. So doing the same as Sysinternals Suite does would have been the desired solution (Store + Winget + Portable version for download via web browser).


More generally speaking:
Please standardize the distribution channels for everything in MS. And not have every department and team use different distribution channels. My suggestion is everything needs to be distributed using the store and winget. And if it is an administrative tool as a standalone that does not require an installation and is ready to use (Like Sysinternals Suite).

I don't want to be constantly looking because every single team does it differently. Also, your software distribution solutions no longer support all of your own applications. Like GPOs still, to this day, cannot deploy MSIX or APPX. And not to mention Office, which has a shitty "deployment tool" that, before it can be used, first needs to be packaged into something else. Doesn't MS have anyone that takes care of compatibility between the teams?

**Comment by SamB** [comment]
> Also note - we have one more store release planned which matches the current release other than pointing folks to https://aka.ms/windbg/download for future updates. We couldn't publish that version before having the download version available.

Okay, I got updated to that version, but when I grab https://aka.ms/windbg/download and try to install, I get the message "A newer version of WinDbg is already installed. To force update, click Update."

So now I don't know whether "update" or not ...

**Comment by dongle-the-gadget** [comment]
The WinDbg version on winget will get outdated as we need to update it frequently (which I don't), whereas the appinstaller file you mentioned will auto-update.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/microsoft/WinDbg-Feedback/issues/91#issuecomment-1512807190

@sylveon, that's available at https://github.com/microsoft/WinDbg-Feedback/issues/264, I believe.

**Comment by sylveon** [comment]
#91 predates #264 ;)

I can close this issue because WinDBG is back in the store (has been for a while actually): https://apps.microsoft.com/detail/9PGJGD53TN86?hl=en-us&gl=CA&ocid=pdpshare
